src/app/CriticalSectionImplPthread.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <cassert>
+#include <pthread.h>
+
+namespace chip {
+namespace Platform {
+
+pthread_mutex_t eventLocker;
+
+void CriticalSectionEnter()
```

Comment:
This locking strategy doesn't make sense to me.  The second concurrent caller to `CriticalSectionEnter` will be calling `pthread_mutex_init` twice on `eventLocker`, which is undefined behavior.

Reply:
yep, I am quickly testing different lock for pthread, it seems nrf presubmit is using  Z MUTEX INITIALIZER for pthread, and other uses pthread, then try to check pthread_mutex_init is common for pthread across platform.....

Reply:
Decide to have the exception for nrf zephyr platform with z mutex initializer. thanks